S/SLEEF/build_tarballs.jl

```diff
@@ -57,14 +58,16 @@ filter!(p -> !(arch(p) == "i686" && Sys.iswindows(p)), platforms) # i686-windows
 # The products that we will ensure are always built
 products = [
     LibraryProduct("libsleef", :libsleef),
+    LibraryProduct("libsleefdft", :libsleefdft),
     LibraryProduct("libsleefquad", :libsleefquad),
+    LibraryProduct("libsleefscalar", :libsleefscalar),
 ]
 
 # Dependencies that must be installed before this package can be built
 dependencies = [
     Dependency(PackageSpec(name="CompilerSupportLibraries_jll", uuid="e66e0078-7015-5450-92f7-15fbd957f2ae")),
-    Dependency("MPFR_jll"),
-    Dependency("OpenSSL_jll"; compat="1.1.10"),
+    Dependency("MPFR_jll"; compat="4.2.1"), # we need 4.2.1 for aarch64-unknown-freebsd
+    Dependency("OpenSSL_jll"; compat="3.0.15"), # we need 3.0.15 for aarch64-unknown-freebsd
```

Comment:
FWIW there is also an OpenSSL 1.1 build for FreeBSD AArch64 so we don't necessarily have to go from 1.1 to 3 here.

For MPFR, if the compat is omitted won't it pick up the latest available version, which would work as well?

Reply:
We definitely want to move as many packages as possible to OpenSSL 3, regardless of aarch64-freebsd :wink:

Reply:
I agree with removing the compat of MPFR though: while in general I prefer specifying compat bounds, this gets tricky for julia stdlibs because they implicitly force compatibility with Julia itself, and MPFR 4.2.1 is only in julia v1.11+

Reply:
Also, @eschnett don't get mad with aarch64-freebsd: if it's too much work to build for that platform just skip it.

Reply:
My main development machine now is an `aarch64`, so I want to support that CPU everywhere possible!

I'll remove the `MPFR` compat bound.

Reply:
And soon you'll be inspired to install FreeBSD on that machine, I'm sure 😄